src/main/java/uk/gov/digital/ho/hocs/casework/api/StageService.java

```diff
@@ -138,40 +138,64 @@ public Stage createStage(UUID caseUUID, CreateStageRequest createStageRequest) {
-
-        UUID currentUserUUID = null;
-        if(maybeActiveStage.isPresent()) {
-            currentUserUUID = maybeActiveStage.get().getUserUUID();
+        Set<Stage> allStages = stageRepository.findAllByCaseUUIDAsStage(caseUUID);
+        Set<Stage> activeStages = allStages.stream().filter(stage -> stage.getTeamUUID() != null).collect(Collectors.toSet());
+        if (activeStages.size() > 1) {
+            log.warn("More than 1 active stage, all active stages will be deactivated");
         }
 
-        // Create new stage
-        Stage newStage = new Stage(caseUUID, createStageRequest.getType(),
-                null, null, createStageRequest.getTransitionNoteUUID());
+        List<Stage> existingStageOfRequestedType = allStages.stream().filter(stage -> stage.getStageType().equals(createStageRequest.getType())).collect(Collectors.toList());
+
+        boolean isRecreation = false;
+        boolean isRecreateOfCurrentlyActive = false;
+        Stage stageToActivate;
+        if (existingStageOfRequestedType.isEmpty()) {
+            log.debug("Stage type of {} does not currently exist for caseUUID: {}, creating new stage",createStageRequest.getType(), caseUUID);
+            stageToActivate = new Stage(caseUUID, createStageRequest.getType(),
+                    null, null, createStageRequest.getTransitionNoteUUID());
+        } else {
+            // Assumes only 1 historical stage existed.
+            log.debug("Stage type of {} exists for caseUUID: {}, recreating stage",createStageRequest.getType(), caseUUID);
+            stageToActivate = existingStageOfRequestedType.get(0);
+            isRecreation = true;
+            isRecreateOfCurrentlyActive = activeStages.contains(stageToActivate);
+        }
 
         CaseData caseData = caseDataService.getCase(caseUUID);
-        calculateDeadlines(newStage, caseData);
+        calculateDeadlines(stageToActivate, caseData);
+
+        stageRepository.save(stageToActivate);
+        log.info("Created Stage: {}, Type: {}, Case: {}, event: {}", stageToActivate.getUuid(), stageToActivate.getStageType(), stageToActivate.getCaseUUID(), value(EVENT, STAGE_CREATED));
+
+        caseDataService.updateCaseData(caseData, stageToActivate.getUuid(), Map.of(CaseworkConstants.CURRENT_STAGE, stageToActivate.getStageType()));
 
-        stageRepository.save(newStage);
-        log.info("Created Stage: {}, Type: {}, Case: {}, event: {}", newStage.getUuid(), newStage.getStageType(), newStage.getCaseUUID(), value(EVENT, STAGE_CREATED));
+        assignTeamAndMemberUserToStage(stageToActivate, createStageRequest.getTeamUUID(), createStageRequest.getUserUUID());
 
-        caseDataService.updateCaseData(caseData, newStage.getUuid(), Map.of(CaseworkConstants.CURRENT_STAGE, newStage.getStageType()));
+        List<UUID> currentUserUUIDList = activeStages.stream().map(BaseStage::getUserUUID).distinct().collect(Collectors.toList());
```

Comment:
Could this be moved to before L192? Also feels like we could do a `findFirst` instead?

Reply:
ok, moved closer to the usage, previously put there as in a later step the userUUID is cleared when the stage is deactivated, I've rearranged some things